src/map/battle.cpp

```diff
@@ -393,7 +393,12 @@ int battle_delay_damage(t_tick tick, int amotion, struct block_list *src, struct
 		damage = 0;
 	}
 
-	if ( !battle_config.delay_battle_damage || amotion <= 1 ) {
+	// If the synchronize damage feature is activated then disabling delay battle damage should only affect skills (for monsters)
+	// If clientamotion is 1 or lower for monsters, we can apply damage directly for normal attacks and don't need to create a timer
+	// Same applies for all types, normal attacks and skills if amotion is 1 or lower
+	if ((!battle_config.delay_battle_damage && (src->type != BL_MOB || !battle_config.synchronize_damage || skill_id != 0))
+		|| (skill_id == 0 && src->type == BL_MOB && battle_config.synchronize_damage && status_get_clientamotion(src) <= 1)
+		|| amotion <= 1) {
```

Comment:
This if is too long again.
It also changes existing behavor.

Reply:
It does not change existing behavior. You'd need to enable the new feature for the behavior to change.

But maybe should just adjust amotion in more simple checks beforehand.

Reply:
Split into 3 if() checks.